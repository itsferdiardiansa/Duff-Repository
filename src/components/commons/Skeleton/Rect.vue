<template>
  <span
    ref="skeletonRectEl"
    :class="`${prefixClass}-skeleton--rect`"
    :style="customStyles"
  />
</template>
<script>
import { computed, ref } from 'vue'

export default {
  props: {
    width: {
      type: [Number, String],
      default: null,
    },
    height: {
      type: [Number, String],
      default: null,
    },
    rounded: {
      type: Boolean,
      default: true,
    },
  },
  setup(props) {
    const skeletonRectEl = ref()

    const customStyles = computed(() => {
      let { width, height, rounded } = props
      let style = {}

      if (width !== null) style.width = width

      if (height !== null) style.height = height

      style.borderRadius = rounded ? '4px' : 0

      return style
    })

    return {
      skeletonRectEl,
      customStyles,
    }
  },
}
</script>
<style lang="scss" scoped>
@keyframes animation-slide {
  0% {
    background-position: -300px center;
  }

  100% {
    background-position: 300px center;
  }
}

.#{$prefixClass}-skeleton--rect {
  display: inline-block;
  height: 22px;
  width: 100%;
  border-radius: 4px;
  background-color: rgb(206, 206, 206);
  background-image: linear-gradient(
    89deg,
    rgb(200, 200, 200),
    rgb(200, 200, 200)
  );
  background-size: 99% 100%;
  background-repeat: no-repeat;
  animation: 1.3s linear 1ms infinite normal backwards running animation-slide;
  animation: 1.3s linear 1ms infinite normal backwards running animation-slide;
}
</style>
