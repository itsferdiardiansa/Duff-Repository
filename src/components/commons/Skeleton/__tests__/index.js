import { mount } from '@vue/test-utils'
import { Rect } from '@common/Skeleton'

describe('common/Skeleton', () => {
  let Component
  let prefixClass

  beforeAll(() => {
    Component = mount(Rect)
    prefixClass = `${Component.rootVM.prefixClass}-skeleton--rect`
  })

  it('will renders ans math snapshot', () => {
    expect(Component.element).toMatchSnapshot()
  })

  it('will render with specific width', async () => {
    const width = '200px'

    await Component.setProps({ width })

    expect(Component.attributes('style')).toContain(width)
    expect(Component.element).toMatchSnapshot()
  })

  it('will render with specific height', async () => {
    const height = '2rem'

    await Component.setProps({ height })

    expect(Component.attributes('style')).toContain(height)
    expect(Component.element).toMatchSnapshot()
  })

  it('will render without being rounded', async () => {
    await Component.setProps({ rounded: false })

    expect(Component.attributes('style')).toContain('border-radius: 0;')
    expect(Component.element).toMatchSnapshot()
  })
})
