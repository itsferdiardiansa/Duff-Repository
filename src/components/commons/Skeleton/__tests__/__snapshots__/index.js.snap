// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Skeleton will render with specific height 1`] = `
<span
  class="lk-skeleton--rect"
  style="border-radius: 4px; width: 200px; height: 2rem;"
/>
`;

exports[`common/Skeleton will render with specific width 1`] = `
<span
  class="lk-skeleton--rect"
  style="border-radius: 4px; width: 200px;"
/>
`;

exports[`common/Skeleton will render without being rounded 1`] = `
<span
  class="lk-skeleton--rect"
  style="border-radius: 0; width: 200px; height: 2rem;"
/>
`;

exports[`common/Skeleton will renders ans math snapshot 1`] = `
<span
  class="lk-skeleton--rect"
  style="border-radius: 4px;"
/>
`;
